### ISSUE TITLE
400 Bad request

### ISSUE THREAD

**Issue by LeroyChen96** [Modern-Treasury/modern-treasury-go#84]
When testing idempotency guarantees, we noticed that the response code does not look right.

Repro steps: Fire two idempotent requests back to back. Not the full code but looks something like:
```
	txn, err := m.client.LedgerTransactions.New(ctx, mt.LedgerTransactionNewParams{
		LedgerEntries: mt.F(entries),
		Description:   mt.F(description),
		ExternalID:    mt.F(coID),
		Status:        mt.LedgerTransactionNewParamsStatusPending,
	}, option.WithHeader("Idempotency-Key", idempotencyKey))
	if err != nil {
		return nil, fmt.Errorf("failed to create transaction: %w", err)
	}

	txn, err = m.client.LedgerTransactions.New(ctx, mt.LedgerTransactionNewParams{
		LedgerEntries: mt.F(entries),
		Description:   mt.F(description),
		ExternalID:    mt.F(coID),
		Status:        mt.LedgerTransactionNewParamsStatusPending,
	}, option.WithHeader("Idempotency-Key", idempotencyKey))
	if err != nil {
		return nil, fmt.Errorf("failed to create transaction: %w", err)
	}
```

We expect the first one to succeed, and the second to return a 409 as per [the documentation](https://docs.moderntreasury.com/platform/reference/idempotent-requests), but instead this is returning us a `400 Bad Request`.
![image](https://github.com/Modern-Treasury/modern-treasury-go/assets/28577511/8ad690ab-4d51-44ca-89cc-577ff4d136ec)

Before we migrated to the official SDK, we were using an unofficially generated SDK that did in fact return us a 409 as expected. We suspect that there might be some additional data massaging happening under the hood that's converting 400+ codes to 400.

**Comment by yjp20** [comment]
Hey, thanks for the detailed bug report again, we found the issue and will push out a fix soon.

**Comment by yjp20** [comment]
Hey @LeroyChen96 , the issue should be fixed in the `next` branch and released soon. Please feel free to try it out and do let us know if it works for you or if there are any issues!

**Comment by andrewlaguna824** [comment]
@yjp20 Is there a timeline to when the `next` will be released?

**Comment by rattrayalex** [comment]
Sorry for the delay there @andrewlaguna824, this has been released in 1.5.0!

**Comment by andrewlaguna824** [comment]
@rattrayalex Thank you!
